packages/compiler/src/server/tspconfig/completion.ts

```diff
@@ -204,10 +319,153 @@ export async function provideTspconfigCompletionItems(
               return item;
             });
             result.push(...enums);
+          } else if (
+            cur.type === "string" &&
+            sourceType === "QUOTE_DOUBLE" &&
+            /{(env\.)?}|{[^{}]*}/g.test(source)
+          ) {
```

Comment:
it's better to handle this kind of completion separated first than mix it up with the schema related resolving, so if the current cursor is at a string value and match the regex, then try to give the list of completion items first. Let's chat about this offline if needed.